### ISSUE TITLE
How can I use this library from another project?

### ISSUE THREAD

**Issue by mbifulco** [kevinzhow/PNChart-Swift#24]
A bit of a noob question, but I'm looking to use the PNChart-Swift library in a swift project I'm working on.  Is there a trick to using is as a framework, or referencing it from an external project?

**Comment by vlafiser** [comment]
Yop, I have the same question. Is it possible without Pod install?

**Comment by mbifulco** [comment]
In the end I went with the Pod install, plus a bridging header.  It seems to work okay for me so far!

**Comment by vlafiser** [comment]
Can you share it for me please? (of course, some example) Thank you so much man! :)

**Comment by mbifulco** [comment]
Ah man I'm sorry, my reply was slightly incorrect - so I ended up using a Podfile that references the **original** PNChart repo, not the swift one.  As far as I can tell, there's no Pod available for the swift library, which might actually be because CocoaPods don't yet allow you to add Swift projects to your workspace.  I'm not sure.  

Anyway, what I did was add the non swift library - this is what I had to do:

There's not much to it, really - two major things are getting the podfile right and installing pods, and then setting up the Bridging header to the Objective-C libaries.

Your podfile should look something like this (for IOS8, of course)

![image](https://cloud.githubusercontent.com/assets/1844496/6527983/138e4aea-c3ed-11e4-90a2-20170dd16633.png)

Once you've got the PNChart project added to your workspace from installing the pods, you'll need a bridging header in your project. This lets Swift interact with the Objective-C classes in the PNChart project.

You can read more about bridging headers here: https://developer.apple.com/library/prerelease/ios/documentation/Swift/Conceptual/BuildingCocoaApps/MixandMatch.html (look for subsection _Importing Objective-C into Swift_)




Enable your bridging header in your project settings:
![image](https://cloud.githubusercontent.com/assets/1844496/6528022/46b9ee92-c3ed-11e4-8f55-b36d8422743a.png)


To get the charts to work, your bridging header should have these lines in it: 

![image](https://cloud.githubusercontent.com/assets/1844496/6527962/022ece1e-c3ed-11e4-8b04-d56ec4bdd5a6.png)


Basically, from there, your project should have access to anything you want in PNCharts.  On the ViewControllers where you use charts, you'll need to import a reference to the PNCharts library:

![image](https://cloud.githubusercontent.com/assets/1844496/6528158/206eba46-c3ee-11e4-9e7f-ce6a183eea3a.png)

From there, everything should be according to the documentation in the PNCharts repo's readme (And their sample projects).

**Comment by vlafiser** [comment]
Wau, THANK YOU for help! I will try it to do this evening. :)

Thank you Mike!!

**Comment by vlafiser** [comment]
Still not working for me. :/

I mean I wrote something bad. If I paste the Obj-C code from example to viewDidLoad function, will work? For me not working. (I know, is it Obj-C, but the bridging know work with it...?)

**Comment by mbifulco** [comment]
Hi @vlafiser - so you're not going to be able to write Obj-C code.  You'll need to translate it to Swift, essentially, which can be a bit of a pain, but is a great learning experience.

For example, my ViewDidLoad() code looks like this for the ViewController in question:

![image](https://cloud.githubusercontent.com/assets/1844496/6543721/075485d2-c4f6-11e4-8497-d9cc1008fe6c.png)


![image](https://cloud.githubusercontent.com/assets/1844496/6543719/efd0772c-c4f5-11e4-9b96-504be261d3dd.png)

**Comment by m-t-space** [comment]
@mbifulco are these directions for JBCharts (https://github.com/Jawbone/JBChartView) or PNCharts?
your examples seem to all reference the JBCharts names...
Or are they that related that the code would be identical?
I'm attempting to set up a new Swift project with this library but am finding it frustrating and confusing. Could anyone lay this out a bit more clearly?

**Comment by mbifulco** [comment]
Wow, yeah, I'm all over the map.  Sorry... disregard me completelt.  I did switch to JB at some point. Sorry @m-t-space  and @vlafiser

**Comment by mbifulco** [comment]
Went back and tried to put together an example using PNChart earlier, and even with a blank Swift project, I'm having trouble getting the cocoapod to connect to my project through the bridging header and compile happily.  I'll play with it a bit more this weekend and let you know if I can get it going (and share my code).

**Comment by iamprem** [comment]
Hey @mbifulco  , Any good?
